test/tbb/test_limiter_node.cpp

```diff
@@ -445,7 +445,14 @@ void test_try_put_without_successors() {
     int try_put_num{3};
     tbb::flow::buffer_node<int> bn(g);
     tbb::flow::limiter_node<int> ln(g, try_put_num);
-    tbb::flow::make_edge(bn, ln);
+#if __GNUC__ && __GNUC__ < 12 && !TBB_USE_DEBUG
+    // Seemingly, GNU compiler generates incorrect code for the call of limiter.register_successor in release (-03)
+    // The function pointer to make_edge workarounds the issue for unknown reason
+    auto make_edge_ptr = tbb::flow::make_edge<int>;
+    make_edge_ptr(bn, ln); //putting the successor back
+#else
```

Comment:
What about unifying this workaround with [this code](https://github.com/oneapi-src/oneTBB/blob/master/test/tbb/test_limiter_node.cpp#L358) by unifying into a separate function?